source/errors.ts

```diff
@@ -45,41 +46,42 @@ export class ReadError extends GotError {
 }
 
 export class ParseError extends GotError {
-	readonly response: Response;
+	readonly response!: Response;
 
 	constructor(error: Error, response: Response, options: NormalizedOptions) {
-		super(`${error.message} in "${format(options as unknown as URL)}"`, error, options);
+		super(`${error.message} in "${format(options as unknown as URL | UrlObject)}"`, error, options);
 		this.name = 'ParseError';
 
 		Object.defineProperty(this, 'response', {
+			enumerable: false,
```

Comment:
Since TS 3.7 with the `useDefineForClassFields` flag on, all class defined properties will be initialized with `enumerable: true`, `configurable: true`, and `writable: true`. I didn't realize I could mitigate this with the declare handler to mitigate that, will fix that in a commit.

Reply:
OK ... so I realized why I didn't do that earlier. Apparently this feature is [broken](https://github.com/microsoft/TypeScript/issues/34972). Should we wait on that, or just add a comment to ignore that for now and move on?

Reply:
> Should we wait on that, or just add a comment to ignore that for now and move on?

Just add a comment that links to https://github.com/microsoft/TypeScript/issues/34972 and move on.